src/api/bulk.ts

```diff
@@ -1213,25 +1216,46 @@ export class QueryJobV2<S extends Schema> extends EventEmitter {
     }
   }
 
-  async getResults(): Promise<Record[]> {
-    try {
-      if (this.#queryResults) {
-        return this.#queryResults;
-      }
+  request<R = unknown>(
+    request: string | HttpRequest,
+    options: Object = {},
+  ): StreamPromise<R> {
+    // if request is simple string, regard it as url in GET method
+    let request_: HttpRequest =
+      typeof request === 'string' ? { method: 'GET', url: request } : request;
 
-      const results = await this.createQueryRequest<Record[] | undefined>({
-        method: 'GET',
-        path: `/${getJobIdOrError(this.jobInfo)}/results`,
-        responseType: 'text/csv',
-      });
+    const httpApi = new HttpApi(this.#connection, options);
+    httpApi.on('response', (response: HttpResponse) => {
+      this.locator = response.headers['sforce-locator']
+    })
+    return httpApi.request<R>(request_);
```

Comment:
`request` method wraps `HttpApi` so that it can access the locator header from the emitted response
https://github.com/jsforce/jsforce/blob/2.0/src/http-api.ts#L113

Reply:
the returned header is `Sforce-Locator` but jsforce converts all headers to lowercase, see: https://github.com/jsforce/jsforce/blob/2.0/src/request.ts#L63